app/models/govspeak_url_checker.rb

```diff
@@ -4,44 +4,63 @@ def initialize govspeak
   end
 
   def find_broken_urls
-    extract_all_urls_from_govspeak.select do |url|
-      checked_url = CheckedUrl.find_or_initialize_by(url: url)
-      if checked_url.new_record? || checked_url.expired?
-        checked_url.ok = check_url(url)
-        checked_url.save!
-      end
+    urls_to_check = extract_all_urls_from_govspeak.select do |url|
+      url = CheckedUrl.find_or_initialize_by(url: url)
+      url.new_record? || url.expired?
+    end
 
-      !checked_url.ok?
+    if urls_to_check.any?
+      broken_urls = check_urls(urls_to_check)
+      urls_to_check.each do |url|
+        CheckedUrl.find_or_initialize_by(url: url) do |checked_url|
+          checked_url.ok = !broken_urls.include?(checked_url.url)
+        end.save!
```

Comment:
If saving immediately we could use http://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by-21 ?